### PR TITLE
let desktop icon sorted by creation time

### PR DESCRIPTION
**Pull request by Feng-Ye520**
Hi,  I want to let desktop icon sorted by creation time on the Mate desktop like the win system, so I modified  the caja.  Please check the code to See if it is appropriate.

### CODE REVIEW

**Comment by lukefromdc** [comment]
I won't be able to test this for function as I have a lot of icons om the
desktop placed by hand and it takes too long to reposition them if they get 
auto-sorted by anything. I can ensure this doesn't take place.when turned
off but that's it

File: src/file-manager/fm-icon-view.c [hunk 1]
```diff
@@ -794,6 +801,16 @@ update_layout_menus (FMIconView *view)
         gtk_action_set_label (action, _("_Organize Desktop by Name"));
     }
 
+	action = gtk_action_group_get_action (view->details->icon_action_group,
+                                          FM_ACTION_CLEAN_UP_BY_TIME);
+    gtk_action_set_sensitive (action, !is_auto_layout);
+
+    if (FM_IS_DESKTOP_ICON_VIEW (view))
+    {
+        gtk_action_set_label (action, _("Organize Desktop by Create Time"));
```
Comment (vkareh) [inline src/file-manager/fm-icon-view.c]: This is inaccurate, `CAJA_FILE_SORT_BY_MTIME` is for Modified Time, not Created Time.

File: src/file-manager/fm-icon-view.h [hunk 2]
```diff
@@ -88,6 +88,7 @@ struct FMIconViewClass
 
     /* Override "clean_up" if your subclass has its own notion of where icons should be positioned */
     void	 (* clean_up)			 (FMIconView *icon_view);
+	void    (* clean_up_by_time)            (FMIconView *icon_view);
```
Comment (vkareh) [inline src/file-manager/fm-icon-view.h]: Indentation here seems off - maybe you're using Tabs where the rest of the code uses Spaces?

File: src/file-manager/fm-icon-view.c [hunk 3]
```diff
@@ -3239,6 +3262,7 @@ fm_icon_view_class_init (FMIconViewClass *klass)
     fm_directory_view_class->widget_to_file_operation_position = fm_icon_view_widget_to_file_operation_position;
 
     klass->clean_up = fm_icon_view_real_clean_up;
+	klass->clean_up_by_time = fm_icon_view_real_clean_up_by_time;
```
Comment (vkareh) [inline src/file-manager/fm-icon-view.c]: Indentaion is off - please follow the existing code's style

**Comment by raveit65** [comment]
Any progress or can this be closed?

**Comment by lukefromdc** [comment]
Closing as no progress since that question asked 6 months ago. Can be reopened if the problems are fixed and the PR updated

**Pull request closed without merging**